command/plugins_install.go

```diff
@@ -300,10 +300,16 @@ func (c *PluginsInstallCommand) InstallFromBinary(opts plugingetter.ListInstalla
 		}})
 	}
 
+	// Remove metadata from plugin path
+	noMetaVersion := semver.Core().String()
```

Comment:
Regarding testing, unit tests for commands are notoriously awful as they involve a lot of moving parts, which if we start changing behaviour starts breaking in completely unrelated places in the code (that's the only reason why the "mock plugins" initialisation method changes in this PR).
I have started looking into this after our (out of band) discussion, but frankly this is quite a disruptive change, and I don't want to delay the release of 1.11.0-alpha2.
I'll propose the following: let's try to merge this tomorrow, and release the alpha, this will let users experiment with a more up-to-date Packer, which ultimately will lead to us figuring out what else needs to be addressed in the code before the final release.
In parallel, I'll work on proper blackbox testing for commands, I had started working on that kind of testing platform in a separate repo, but I do believe we can make things work in a semi self-contained way. I'll prioritise this effort since it'll increase confidence in those changes, and will give us a base to improve testing all around later.

Does that sound good to you? If so could you approve this PR tomorrow so we can merge and release; otherwise please let me know what would be acceptable for you, and we can discuss the next steps towards merging this PR.

Reply:
> especially related to the addition of the InstallList.ApiVersion

OK nevermind I'm an idiot, is this related to the sorting tests? That can be unit tested for sure, sorry I misunderstood your point! I'll reroll that change with those changes in.

Reply:
And good call, `semver` doesn't work with API versions, that's broken.
I'll fix that and add the tests right now, thanks for the persistence and the callout @sylviamoss

Reply:
Should be fixed now.

Reply:
I think you still talked through a point that I was also suggesting to be tested here, so your first message still makes sense, and for that I am with you on your decision if that's the best way to go. 

For the InstallList.ApiVersion I did notice unit tests that could have been updated to cover that code and I am glad you caught a bug because of them 😅  thanks for working on them!